catroid/src/main/java/org/catrobat/catroid/content/actions/RepeatAction.kt

```diff
@@ -0,0 +1,108 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.content.actions
+
+import android.util.Log
+import com.badlogic.gdx.scenes.scene2d.actions.RepeatAction
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.formulaeditor.InterpretationException
+
+class RepeatAction : RepeatAction() {
+    private var executedCount = 0
+    private var repeatCount: Formula? = null
+    private var sprite: Sprite? = null
+    private var isCurrentLoopInitialized = false
+    private var isRepeatActionInitialized = false
+    private var repeatCountValue = 0
+    private var currentTime = 0f
+    private var isForeverRepeat = false
+    public override fun delegate(delta: Float): Boolean {
+        if (!isRepeatActionInitialized) {
+            isRepeatActionInitialized = true
+            interpretRepeatCount()
+        }
+        if (!isCurrentLoopInitialized) {
+            currentTime = 0f
+            isCurrentLoopInitialized = true
+        }
+        currentTime += delta
+        if (repeatCountValue < 0) {
+            repeatCountValue = 0
+        }
+        if (executedCount >= repeatCountValue && !isForeverRepeat) {
+            return true
+        }
+        if (action != null && action.act(delta) && currentTime >= LOOP_DELAY) {
+            executedCount++
+            if (executedCount >= repeatCountValue && !isForeverRepeat) {
+                return true
+            }
+            isCurrentLoopInitialized = false
+            if (action != null) {
+                action.restart()
+            }
+        }
+        return false
+    }
+
+    override fun restart() {
+        isCurrentLoopInitialized = false
+        isRepeatActionInitialized = false
+        executedCount = 0
+        super.restart()
+    }
+
+    private fun interpretRepeatCount() {
+        try {
+            val interpretation =
+                if (repeatCount == null) java.lang.Double.valueOf(0.0) else repeatCount!!.interpretDouble(
+                    sprite
+                )
+            repeatCountValue = interpretation.toInt()
+        } catch (interpretationException: InterpretationException) {
+            repeatCountValue = 0
+            Log.d(
+                javaClass.simpleName,
+                "Formula interpretation for this specific Brick failed.",
+                interpretationException
+            )
+        }
+    }
+
+    fun setSprite(sprite: Sprite?) {
+        this.sprite = sprite
+    }
+
+    fun setIsForeverRepeat(isForeverRepeat: Boolean) {
+        this.isForeverRepeat = isForeverRepeat
+    }
+
+    fun setRepeatCount(repeatCount: Formula?) {
+        this.repeatCount = repeatCount
```

Comment:
Since the setter is public, why not change `private var` to `var`? Kotlin will create implicit functions to interact with Java.

For other cases, where you only need a public getter, you can also declare it as `var `and declare the setter as private. See https://kotlinlang.org/docs/reference/properties.html under setterVisibility.